x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -308,6 +310,7 @@ func TestInput(t *testing.T) {
 				config["request.url"] = server.URL
 				t.Cleanup(server.Close)
 				t.Cleanup(func() { timeNow = time.Now })
+				defer os.RemoveAll(filepath.Join(os.TempDir(), "logs"))
```

Comment:
Are you expecting this to run when the test is finished?

Reply:
Expecting to run at the end of the testcase. But not 100% sure if that is the case

Reply:
This will run at the start of the test case when the test server constructor returns.

Reply:
Which means the directory is not yet created there at that time and the removeAll was never removing things? I will try to use `t.TempDir()` to see if the error still persists after we close the Tracer.

Reply:
I think with the addition of the context-dependent lumberjack close this all should become moot and you'll be able to go back to using the testing temp and cleanup.